src/js/angular/core/directives/queryeditor/query-editor.controller.js

```diff
@@ -23,17 +23,14 @@ function QueryEditorCtrl($scope, $timeout, toastr, $repositories, $modal, ModalS
         sameAs: true
     };
 
-    let principal = $jwtAuth.getPrincipal();
+    let principal;
+    $scope.getPrincipal = function() {
+        return Promise.resolve($jwtAuth.getPrincipal()).then((response) => principal = response);
```

Comment:
I don't like this very much. IMO it'd be better if the $jwtAuth.getPrincipal() method is refactored to return a Promise in either case in order to not force method clients to either check if the result is a promise or not and to wrap it in later scenario. It's sloppy implementation.

Reply:
Refactored it. Now `jwtAuth.getPrincipal()` always returnes a promise. If the principal is already fetched it will return it wrapped in a promise. If not, it will return a promise which will resolve once secutiry is initialized (there is an event thrown then).